src/hooks/usePinata/usePinata.ts

```diff
@@ -1,38 +1,79 @@
 import useSubmit from "hooks/useSubmit"
 import useToast from "hooks/useToast"
+import { useCallback } from "react"
+import { useFormContext } from "react-hook-form"
+import getRandomInt from "utils/getRandomInt"
 import pinFileToIPFS, {
-  PinataPinFileResponse,
   PinToIPFSProps,
+  PinataPinFileResponse,
 } from "./utils/pinataUpload"
 
 type Props = Partial<{
   onSuccess: (data: PinataPinFileResponse) => void
   onError: (error: any) => void
+  fieldToSetOnSuccess: string
+  fieldToSetOnError: string
 }>
 
 export type Uploader = {
   onUpload: (data?: PinToIPFSProps) => void
   isUploading: boolean
 }
 
-const usePinata = ({ onError, onSuccess }: Props = {}): Uploader => {
+const submit = (ipfsProps: PinToIPFSProps) => pinFileToIPFS(ipfsProps)
```

Comment:
Couldn't we just pass `pinFileToIPFS` to `useSubmit` below? 👀

Reply:
Oh, that is a nice catch, we should be able to pass that, let me try that out!

Reply:
It seems to work as expected! Changed in https://github.com/guildxyz/guild.xyz/pull/1199/commits/3d35dbb3895eb39fb6094c7da43008b896eee0cf